### ISSUE TITLE
Better Error Handling

### ISSUE THREAD

**Issue by Przeszaf** [braintree/braintree_ios#1152]
### Is your feature request related to a problem? Please describe.

Hello! 

I use BrainTree for 3D Secure Actions. Unfortunately, when user cancels the flow on `Secure Checkout` window.

![Simulator Screenshot - iPhone 15 - 2023-12-11 at 10 42 47](https://github.com/braintree/braintree_ios/assets/25778302/139ede02-dd88-4f8f-b408-c6e2e728bca4)

I get `Error Domain=com.braintreepayments.BTThreeDSecureFlowErrorDomain Code=5 "(null)"`.

I'd expec one of `BTThreeDSecureError` to be thrown - i.e. `BTThreeDSecureError.cancelled`.

Not sure if there is anything wrong I'm doing. Here is the code - card is tokenized by our server.

```
        let btRequest = BTThreeDSecureRequest(payload: verifyCardPayload)
        let btClient = BTAPIClient(authorization: action.data.clientToken)!
        btThreeDSecureClient = BTThreeDSecureClient(apiClient: btClient)
        btRequest.threeDSecureRequestDelegate = self
        btThreeDSecureClient?.startPaymentFlow(btRequest) { result, error in
            if let error {
                print(error)
            } else if let nonce = result?.tokenizedCard?.nonce {
                print(nonce)
            } else {
                print(result)
            }
        }
```

Where `verifyCardPayload` is:

```
  ▿ amount : Optional<String>
    - some : "0.01"
  ▿ nonce : Optional<String>
    - some : "tokencc_xyz"
  ▿ bin : Optional<String>
    - some : "123456"
  ▿ email : Optional<String>
    - some : "a@b.com"
  - dfReferenceId : nil
  - exemptionRequested : false
  - challengeRequested : false
  - billingAddress : nil
  - additionalInformation : nil
```

Let me know if you need any more information. I use version `6.10.0`

**Comment by scannillo** [comment]
👋 Hello @Przeszaf - thank you for using the Braintree iOS SDK and opening this issue with us. This is really helpful feedback. This looks like a combination of a bug and a limitation in our current error API design.

For the bug, I made a [quick fix on this branch](https://github.com/braintree/braintree_ios/compare/main...fix-3ds-cancel-description) which populates the `localizedDescription` correctly for the 3DS cancellation case. We can have that in PR soon after we update a few unit tests.

For the API limitation/design, it relates to [this previous GH issue](https://github.com/braintree/braintree_ios/issues/1080). Since we're supporting Swift & Objective-C, you would currently have to cast to `NSError` to access the Int value of the error code (for ex: `(error as! NSError).code`). We list all of the SDK error codes in a markdown file [here](https://github.com/braintree/braintree_ios/blob/main/SDK_ERROR_CODES.md). Ideally instead, you can more easily compare error codes to values that the SDK provides. This is something we are looking to improve in a future version of the SDK and welcome any and all feedback!

**Comment by scannillo** [comment]
DTBTSDK-3276 for internal tracking

**Comment by jaxdesmarais** [comment]
Hey @Przeszaf -

The fix for both the 3DS cancel case and making the errors public was released in [version 6.11.0](https://github.com/braintree/braintree_ios/releases/tag/6.11.0).
